odc/stats/plugins/l34_utils/l4_bare_gradation.py

```diff
@@ -7,10 +7,18 @@
 
 def bare_gradation(xx: xr.Dataset, bare_threshold, veg_cover):
 
+    bs_pc_50 = expr_eval(
+        "where((a!=a), nodata, a)",
+        {"a": xx.bs_pc_50.data},
+        name="mark_veg",
+        dtype="float32",
+        **{"nodata": NODATA},
+    )
+
     # Map any data > 100 ---> 100
     bs_pc_50 = expr_eval(
         "where((a>100)&(a!=nodata), 100, a)",
-        {"a": xx.bs_pc_50.data},
+        {"a": bs_pc_50},
         name="mark_veg",
         dtype="uint8",
         **{"nodata": NODATA},
```

Comment:
no need for this block, could you also give dask task a meaningful name, in general? otherwise it'd be very hard to locate the issue if anything wrong.

Reply:
Addressed!